examples/lighting-app/nrfconnect/prj.conf

```diff
@@ -29,7 +29,7 @@ CONFIG_OPENTHREAD_NETWORK_NAME="OpenThread"
 CONFIG_OPENTHREAD_XPANID="11:11:11:11:22:22:22:22"
 
 # Bluetooth overrides
-CONFIG_BT_DEVICE_NAME="ChipLight"
+CONFIG_BT_DEVICE_NAME="MatterLight"
```

Comment:
Quick question: why the device name on any advertisement from Matter reference implementation samples? For debugging?

The spec doesn't call for any device name in adv, and it calls for the entire Matter announcement to fit in the advertisement without needing a scan response:

```
In order to reduce 2.4 GHz spectrum congestion due to active BLE scanning, and to extend battery life in battery-powered devices, all critical data used for commissionable CHIP device discovery is contained in the Advertising Data rather than the Scan Response Data. This allows a BLE Commissioner to passively scan (i.e., not issue Scan Requests upon receiving scannable advertisements) and still be able to receive all information needed to commission a device.

Note that if additional vendor-specific information is to be conveyed and does not fit within the Advertising Data, it may be included in the Scan Response Data. See <<Manufacturer-specific data>> for details on including vendor-specific information.
```

Showing the advertisement of device name in examples, especially telling what kind it is, can give the impression it is needed, when it was designed specifically so that privacy was preserved about product types (e.g. VID/PID is not mandatory either)

Reply:
Yes, for debugging. It's much easier to analyze the traffic with the BLE sniffer etc. when you can easily filter by the device name. We don't use the scan response data. The BLE name fits in the advertising data if it's short enough. It's a good point, however, that we should guard the code for adding the device name to the advertising data with some compile time constant.